### PR TITLE
Add tests to check for incomplete form submissions

### PR DESCRIPTION
**Pull request by techncl**

### CODE REVIEW

File: test/controllers/TransferSummaryControllerSpec.scala [hunk 1]
```diff
@@ -151,11 +151,83 @@ class TransferSummaryControllerSpec extends FrontEndTestHelper {
...
+
+      val controller = new TransferSummaryController(getAuthorisedSecurityComponents, new GraphQLConfiguration(app.configuration),
+        getValidKeycloakConfiguration, consignmentService, exportService(app.configuration), langs)
+      val finalTransferConfirmationSubmitResult = controller.finalTransferConfirmationSubmit(consignmentId)
+        .apply(FakeRequest(POST, s"/consignment/$consignmentId/transfer-summary")
+          .withFormUrlEncodedBody(finalTransferConfirmationForm(openRecordsValue = true, transferLegalOwnershipValue = false): _*)
+          .withCSRFToken)
 
       playStatus(finalTransferConfirmationSubmitResult) mustBe BAD_REQUEST
+      print(contentAsString(finalTransferConfirmationSubmitResult))
```
Comment (TomJKing) [inline test/controllers/TransferSummaryControllerSpec.scala:192]: Is there a reason for this `print()` statement?

File: test/controllers/TransferSummaryControllerSpec.scala [hunk 2]
```diff
@@ -151,11 +151,83 @@ class TransferSummaryControllerSpec extends FrontEndTestHelper {
...
+      contentAsString(finalTransferConfirmationSubmitResult) must include("govuk-error-message")
+      contentAsString(finalTransferConfirmationSubmitResult) must include("error")
+      contentAsString(finalTransferConfirmationSubmitResult) must include("There is a problem")
+      contentAsString(finalTransferConfirmationSubmitResult) must include("#error-transferLegalOwnership")
+      contentAsString(finalTransferConfirmationSubmitResult) must include("transferSummary.transferLegalOwnership.error")
```
Comment (TomJKing) [inline test/controllers/TransferSummaryControllerSpec.scala:197]: Might me worth adding a check to make sure the other error is not displayed

File: test/controllers/TransferSummaryControllerSpec.scala [hunk 3]
```diff
@@ -151,11 +151,83 @@ class TransferSummaryControllerSpec extends FrontEndTestHelper {
...
+
+      val controller = new TransferSummaryController(getAuthorisedSecurityComponents, new GraphQLConfiguration(app.configuration),
+        getValidKeycloakConfiguration, consignmentService, exportService(app.configuration), langs)
+      val finalTransferConfirmationSubmitResult = controller.finalTransferConfirmationSubmit(consignmentId)
+        .apply(FakeRequest(POST, s"/consignment/$consignmentId/transfer-summary")
+          .withFormUrlEncodedBody(finalTransferConfirmationForm(openRecordsValue = false, transferLegalOwnershipValue = true): _*)
+          .withCSRFToken)
+
+      playStatus(finalTransferConfirmationSubmitResult) mustBe BAD_REQUEST
+      print(contentAsString(finalTransferConfirmationSubmitResult))
```
Comment (TomJKing) [inline test/controllers/TransferSummaryControllerSpec.scala:224]: Is there a reason for the `print` statement?

File: test/controllers/TransferSummaryControllerSpec.scala [hunk 4]
```diff
@@ -151,11 +151,83 @@ class TransferSummaryControllerSpec extends FrontEndTestHelper {
...
+
       contentAsString(finalTransferConfirmationSubmitResult) must include("govuk-error-message")
       contentAsString(finalTransferConfirmationSubmitResult) must include("error")
+      contentAsString(finalTransferConfirmationSubmitResult) must include("There is a problem")
+      contentAsString(finalTransferConfirmationSubmitResult) must include("#error-openRecords")
+      contentAsString(finalTransferConfirmationSubmitResult) must include("transferSummary.openRecords.error")
```
Comment (TomJKing) [inline test/controllers/TransferSummaryControllerSpec.scala:230]: Might be worth adding another check to make sure the other error is not displayed.

File: test/controllers/TransferSummaryControllerSpec.scala [hunk 5]
```diff
@@ -151,11 +151,83 @@ class TransferSummaryControllerSpec extends FrontEndTestHelper {
...
+      contentAsString(finalTransferConfirmationSubmitResult) must include("error")
+      contentAsString(finalTransferConfirmationSubmitResult) must include("There is a problem")
+      contentAsString(finalTransferConfirmationSubmitResult) must include("#error-openRecords")
+      contentAsString(finalTransferConfirmationSubmitResult) must include("#error-transferLegalOwnership")
+      contentAsString(finalTransferConfirmationSubmitResult) must include("transferSummary.openRecords.error")
+      contentAsString(finalTransferConfirmationSubmitResult) must include("transferSummary.transferLegalOwnership.error")
+    }
+
+    "display correct error when only the open records option is selected and the final transfer confirmation form is submitted" in {
+      val client = new GraphQLConfiguration(app.configuration).getClient[gcs.Data, gcs.Variables]()
```
Comment (TomJKing) [inline test/controllers/TransferSummaryControllerSpec.scala:169]: You could abstract the setup of the test into a separate function, as it is the same for both the tests

**Review by TomJKing — COMMENTED** [review]
Looks good. Just a couple of minor comments/questions

File: test/controllers/TransferSummaryControllerSpec.scala (hunk 1 above)
Comment (techncl) [reply test/controllers/TransferSummaryControllerSpec.scala:192]: Ah, yes. Sorry, I forgot to remove them 😬

**Review by TomJKing — APPROVED** [review]
Looks good.

**Pull request merged**